### PR TITLE
feat: add type field to user highlights table

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Adds the `type` field to the `user_highlights` table
- Adds the type field to the database migrations
- Adds the type field to the user highlight entity

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes https://github.com/open-sauced/api/issues/231

## Mobile & Desktop Screenshots/Recordings

Sample output

```json
    {
      "id": 166,
      "user_id": 37635083,
      "url": "https://github.com/open-sauced/insights/pull/1429",
      "title": "Improving the overall style ",
      "highlight": "This commit adds width and max width for the \"created at\" element, improving the overall style and design. The changes aim to enhance the visual appearance and ensure better alignment.",
      "type": "pull_request",
      "pinned": false,
      "featured": false,
      "created_at": "2023-07-26T00:54:55.138Z",
      "updated_at": "2023-07-26T00:54:55.138Z",
      "deleted_at": null,
      "shipped_at": "2023-07-26T05:54:55.082Z",
      "name": "Nechiforel David-Samuel (NsdHSO)",
      "login": "NsdHSO"
    },
```

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: migrations/20230405116001_add_user_highlights.sql [hunk 1]
```diff
@@ -13,6 +13,7 @@ create table public.user_highlights
   -- elastic columns
   title character varying(100) collate pg_catalog."default" not null default '',
   url character varying(255) collate pg_catalog."default" not null default '',
+  type character varying(25) collate pg_catalog."default" not null default 'pull_request',
```
Comment (jpmcb) [inline migrations/20230405116001_add_user_highlights.sql:16]: Does this need to be in a separate migration, not added to a previous one? I guess I'm not super aware of how we've been applying migrations or the operation around getting from the bare DB to all migrations applied. In the past, I've applied migrations by date (like we have in the `migrations/` directory, but it was always roll forward migrations (i.e., no new bits to old migrations)

Comment (brandonroberts) [reply migrations/20230405116001_add_user_highlights.sql:16]: Good point. We've been adding them to the current migrations, but I agree having it as a separate migration would have better structure going forward.

Comment (brandonroberts) [reply migrations/20230405116001_add_user_highlights.sql:16]: We should have a better process around applying the most recent migrations also

File: src/user/entities/user-highlight.entity.ts [hunk 2]
```diff
@@ -59,6 +59,13 @@ export class DbUserHighlight extends BaseEntity {
   @Column("text")
   public highlight: string;
 
+  @ApiModelProperty({
+    description: "Highlight type",
+    example: "pull_request",
+  })
+  @Column("text")
+  public type: string;
```
Comment (OgDev-01) [inline src/user/entities/user-highlight.entity.ts:67]: are we allowing any string to be passed for the highlights type? 

I was thinking we needed an explicit union of `"pull_request" | "blog_post" | "issue" | "milestone"`

Comment (brandonroberts) [reply src/user/entities/user-highlight.entity.ts:67]: Yes, we can validate against certain types. I'll add it here

https://github.com/open-sauced/api/blob/beta/src/user/dtos/create-user-highlight.dto.ts

**Review by OgDev-01 — APPROVED** [review]
🍕

**Review by bdougie — APPROVED** [review]

**Pull request merged**